### DISCUSSION TITLE
Registering dynamically generated form elements in JSX object?

### DISCUSSION THREAD

**Discussion by Heilemann** [react-hook-form/react-hook-form#8085] [Q&A] [answered]
Long story short, I am generating html from handlebars and converting it to a JSX object which I inject into my document.

The handbars template will include form elements and I want to use react-hook-form to register, so I recursively loop over the JSX object's children to register the field. But it doesn't work.

Here's how I inject it:

<img width="470" alt="CleanShot 2022-03-23 at 17 19 15@2x" src="https://user-images.githubusercontent.com/203049/159797462-42588a15-0cc0-4030-9312-6710f948b2f6.png">

My onChange watch works (I've tested), but this way of injecting the register function doesn't result in the fields touched triggering the watch function.

Any ideas for how I might do this?

**Comment by bluebill1049** [comment]
Could you share a simple reproduce csb?

**Reply by Heilemann** [reply]
Yep, absolutely. [Here you go](https://codesandbox.io/s/morning-pond-1y2d26?file=/src/App.js)!

The input on the left is the dynamically registered one, it doesn't work. The one on the right is done in the normal way, and it works. When the working one fires the change event it logs the values object, which includes the other inputs name with an undefined value.

**Reply by Heilemann** [reply]
@bluebill1049 Any thoughts? I'm wondering if it isn't because the component updates, and I need to re-register? I'll try to poke at that a bit when I've got a moment.

**Comment by Heilemann** [comment, marked as answer]
I found the solution. I was of course adding the register the my JSX elements, but I never returned those changed elements to be used properly. My bad!

**Reply by bluebill1049** [reply]
sorry, i was too busy with stuff yesterday. glad you work it out.
